libtenzir/src/tql2/eval_binary.cpp

```diff
@@ -731,36 +731,39 @@ auto eval_and_or(evaluator& self, const ast::binary_expr& x) -> multi_series {
     } else {
       static_assert(detail::always_false_v<decltype(Op)>, "unsupported op");
     }
-    const auto eval_right = [&](int64_t start, int64_t end) -> multi_series {
-      return map_series(
-        self.slice(left_begin + start, left_begin + end).eval(x.right),
-        [&](series right) -> multi_series {
-          if (is<bool_type>(right.type)) {
-            return right;
-          }
-          if (not is<null_type>(right.type)) {
-            diagnostic::warning("expected `bool`, but got `{}`",
-                                right.type.kind())
-              .primary(x.right)
-              .emit(self.ctx());
-          }
-          return series::null(bool_type{}, right.length());
-        });
-    };
     TENZIR_ASSERT(typed_left);
     const auto get_left = [&](int64_t i) -> bool {
       return typed_left->array->IsValid(i) and typed_left->array->GetView(i);
     };
-    auto result = multi_series{};
+    auto builder = arrow::BooleanBuilder{};
+    check(builder.Reserve(self.length()));
```

Comment:
Should this be setup even outside of the `map_series`, so directly at the top of the function? That way we not only introduce no additional splits, but can also unify splits that occurred before.

Reply:
Great idea. Implemented this here: 40dbc36fbc3